File: cassidy/js_code/producer.js
const { Kafka } = require("kafkajs")

run();

async function run() {
    try {
        const kafka = new Kafka({
            "clientId": "lab2",
            "brokers": ["localhost:9093", "localhost:9094", "localhost:9095"]
        });

        const producer = kafka.producer();
        console.log("Trying to connect ...");
        await producer.connect();
        console.log("Connected");

        const result = await producer.send({
            "topic": "topic2",
            "messages": [{
                "value": "ayoo",
                "partition": 0
            }]
        });

        console.log(`Message sent: ${JSON.stringify(result)}`);

        await producer.disconnect();

    } catch(e) {
        console.error(e);
    } finally {
        process.exit(0);
    }
}

File: cassidy/python_code/producer.py
from kafka import KafkaProducer

producer = KafkaProducer(
    bootstrap_servers=['localhost:9093', 'localhost:9094', 'localhost:9095'])

print("Connected!")

for i in range(10):
    part = i % 3
    if i % 2 == 0:
        producer.send(
            topic='topic',
            value=b'bla bla',
            partition=part
        )
    else:
        producer.send(
            topic='topic2',
            key=i.to_bytes(2, 'big'),
            value=b'Big blobbb',
            partition=part
        )
    print("Message sent!")

producer.close()
